File: src/com/colacelli/ircbot/plugins/nickserv/NickServPlugin.java
package com.colacelli.ircbot.plugins.nickserv;

import com.colacelli.ircbot.IRCBot;
import com.colacelli.ircbot.Plugin;
import com.colacelli.irclib.actors.User;
import com.colacelli.irclib.connection.listeners.OnConnectListener;
import com.colacelli.irclib.messages.PrivateMessage;

public class NickServPlugin implements Plugin {
    private String password;
    private OnConnectListener listener;

    public NickServPlugin(String password) {
        this.password = password;

        listener = (connection, server, user) -> {
            User.Builder userBuilder = new User.Builder();
            userBuilder.setNick("nickserv");

            PrivateMessage.Builder builder = new PrivateMessage.Builder();
            builder
                    .setSender(connection.getUser())
                    .setReceiver(userBuilder.build())
                    .setText("identify " + password);

            connection.send(builder.build());
        };
    }

    @Override
    public String getName() {
        return "NICKSERV";
    }

    @Override
    public void onLoad(IRCBot bot) {
        if (password != null && !password.isEmpty()) {
            bot.addListener(listener);
        }
    }

    @Override
    public void onUnload(IRCBot bot) {
        bot.removeListener(listener);
    }
}

File: src/com/colacelli/ircbot/plugins/nickserv/NickServPlugin.kt
package com.colacelli.ircbot.plugins.nickserv

import com.colacelli.ircbot.IRCBot
import com.colacelli.ircbot.base.Plugin
import com.colacelli.irclib.actors.User
import com.colacelli.irclib.connection.Connection
import com.colacelli.irclib.connection.Server
import com.colacelli.irclib.connection.listeners.OnConnectListener
import com.colacelli.irclib.messages.PrivateMessage

class NickServPlugin(password: String) : Plugin {
    val listener = object : OnConnectListener {
        override fun onConnect(connection: Connection, server: Server, user: User) {
            return connection.send(PrivateMessage(
                    "identify $password",
                    connection.user,
                    User("NickServ")
            ))
        }
    }
    override var name = "nickserv"

    override fun onLoad(bot: IRCBot) {
        bot.addListener(listener)
    }

    override fun onUnload(bot: IRCBot) {
        bot.removeListener(listener)
    }
}